src/client/login/index.tsx

```diff
@@ -107,20 +96,25 @@ const useStyles = makeStyles((theme) =>
   }),
 )
 
-const LoginPage = ({
+const LoginPage: FunctionComponent<LoginPageProps> = ({
   location,
-  isLoggedIn,
-  getOTPEmail,
-  getEmailValidator,
-  verifyOTP,
-  setEmail,
-  setOTP,
-  email,
-  emailValidator,
-  variant,
-  setLoginInfoMessage,
 }) => {
   const classes = useStyles()
+  const dispatch = useDispatch()
+  const getEmailValidator = dispatch(loginActions.getEmailValidationGlobExpression())
+  const setLoginInfoMessage = (message:string) => dispatch(rootActions.setInfoMessage(message))
+  const emailValidator = useSelector(
+    (state: GoGovReduxState) => state.login.emailValidator
+  )
+  const isLoggedIn = useSelector(
+    (state: GoGovReduxState) => state.login.isLoggedIn
+  )
+  const email = useSelector(
+    (state: GoGovReduxState) => state.login.email
+  )
+  const variant: VariantType = useSelector(
+    (state: GoGovReduxState) => state.login.formVariant
+  )
```

Comment:
Possible to do something like...
```suggestion
  const {
    emailValidator,
    isLoggedIn,
    email,
    formVariant: variant,
  } = useSelector(
    (state: GoGovReduxState) => state.login
  )
```